tests/test_core.py

```diff
@@ -1049,8 +1109,11 @@ def test_generate_with_seeding(atol=1e-4, rtol=1e-8):
     num_generators = 3
     for seed in seeds:
         generators = []
-        for i in range(num_generators):
+        for i in range(num_generators - 1):
             generators.append(_create_scaper_with_random_seed(seed))
+        
+        generators.append(_create_scaper_with_random_seed(seed))
+        generators[-1].set_random_state(seed)
```

Comment:
Since unit tests should be as modular/small/simple as possible, please let's break this out into a separate `test_set_random_state()`